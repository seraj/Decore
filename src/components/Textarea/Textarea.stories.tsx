import { boolean, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Textarea from "./Textarea";

const stories = storiesOf("Components/Textarea", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <Textarea
      dark={boolean("dark", false)}
      placeholder="Write something here..."
    ></Textarea>
  ))
);
