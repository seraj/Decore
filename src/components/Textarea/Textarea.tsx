import * as React from "react";
import StyledTextarea from "./styled/StyledTextarea";
import TextareaProps from "./Textarea.props";

const Textarea: React.FC<TextareaProps> = ({ children, ...props }) => (
  <StyledTextarea {...props}>{children}</StyledTextarea>
);

export default Textarea;
