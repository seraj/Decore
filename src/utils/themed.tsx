import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
