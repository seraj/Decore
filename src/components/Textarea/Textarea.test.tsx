import * as React from "react";

import themed from "../../utils/themed";
import Textarea from "./Textarea";

describe("Textarea", () => {
  it("should render correctly", () => {
    const component = themed(<Textarea>Test</Textarea>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
