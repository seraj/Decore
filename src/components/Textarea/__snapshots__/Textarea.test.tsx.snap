// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should render correctly 1`] = `
.c0 {
  border: 1.5px solid #999999;
  padding: 0.6em 0.8em;
  color: #333333;
}

<textarea
  className="c0"
>
  Test
</textarea>
`;
